dnsmasq: add append_conf_file() to dnsmasq.init

codemarauder opened the pull request:
Added a function to append additional conf files in dnsmasq.conf format which can be configured in /etc/config/dhcp as:

list conf_file "/path/to/additional-conf"

The function uses xappend and append_extramount to add config files to the config file in /var/etc/dnsmasq-*.conf

Compile tested: x86, mt7621, ath79
Run tested: x86, mt7621, ath79

Signed-off-by: Nishant Sharma <nishant@hopbox.in>

codemarauder committed:
dnsmasq: add append_conf_file() to dnsmasq.init

Added a function to append additional conf files in dnsmasq.conf format which can be configured in /etc/config/dhcp as:

list conf_file "/path/to/additional-conf"

The function uses xappend and append_extramount to add config files to the config file in /var/etc/dnsmasq-*.conf

Compile tested: x86, mt7621, ath79
Run tested: x86, mt7621, ath79

Signed-off-by: Nishant Sharma <nishant@hopbox.in>

dave14305 commented:
You can already add additional conf files in `/tmp/dnsmasq.d` or use the `extraconftext` option to add dnsmasq options to the main config. What’s the benefit of another method?

Lastly, it doesn’t look like you read the parameter or call your function anywhere.

codemarauder committed:
dnsmasq: add function call to append_conf_file() in dnsmasq.init

Add the line to call function append_conf_file() to parse following uci
config:

```
list conf_file '/path/to/additional_conf'
```

Signed-off-by: Nishant Sharma <nishant@hopbox.in>

HSXX commented:
dnsmasq already have a option confdir.
